compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -627,13 +746,22 @@ abstract class GenJSCode extends plugins.PluginComponent
                 withNoinline(shouldMarkNoinline)
 
             val methodDef = {
-              if (sym.isClassConstructor) {
-                js.MethodDef(static = false, methodIdent,
+              if (isJSClassConstructor) {
+                currentMethodInfoBuilder.setIsExported(true)
```

Comment:
Oh, I see. The constructor method needs to be exported since it needs to be added to the prototype of the class itself and can be retrieved via introspection in JavaScript. So if the class is instantiated, the ctor must be retained.